src/llama_stack/providers/utils/memory/openai_vector_store_mixin.py

```diff
@@ -85,14 +89,56 @@ def __init__(
         self,
         files_api: Files | None = None,
         kvstore: KVStore | None = None,
+        vector_stores_config: VectorStoresConfig | None = None,
     ):
         self.openai_vector_stores: dict[str, dict[str, Any]] = {}
         self.openai_file_batches: dict[str, dict[str, Any]] = {}
         self.files_api = files_api
         self.kvstore = kvstore
+        self.vector_stores_config = vector_stores_config
         self._last_file_batch_cleanup_time = 0
         self._file_batch_tasks: dict[str, asyncio.Task[None]] = {}
 
+    async def _rewrite_query_for_search(self, query: str) -> str:
+        """Rewrite a search query using the configured LLM model for better retrieval results."""
+        if (
+            not self.vector_stores_config
+            or not self.vector_stores_config.rewrite_query_params
+            or not self.vector_stores_config.rewrite_query_params.model
+        ):
+            raise ValueError(
+                "Query rewriting requested but not configured. Please configure rewrite_query_params.model in vector_stores config."
+            )
+
+        model = self.vector_stores_config.rewrite_query_params.model
+        model_id = f"{model.provider_id}/{model.model_id}"
+
+        # Use custom prompt from config if provided, otherwise use built-in default
+        # Users only need to configure the model - prompt is automatic with optional override
+        custom_prompt = self.vector_stores_config.rewrite_query_params.prompt
+        if custom_prompt:
+            if "{query}" not in custom_prompt:
+                raise ValueError("'{query}' is required in the custom_prompt")
+
+            prompt = custom_prompt.format(query=query)
+        else:
+            prompt = DEFAULT_QUERY_REWRITE_PROMPT.format(query=query)
+
+        request = OpenAIChatCompletionRequestWithExtraBody(
+            model=model_id,
+            messages=[OpenAIUserMessageParam(role="user", content=prompt)],
+            max_tokens=self.vector_stores_config.rewrite_query_params.max_tokens or 100,
+            temperature=self.vector_stores_config.rewrite_query_params.temperature or 0.3,
+        )
+
+        response = await self.inference_api.openai_chat_completion(request)  # type: ignore
```

Comment:
openai_chat_completion can throw exceptions. they may include config details of the service that should not be exposed to users, e.g. the model being used, api credentials. a safer approach is to catch and log the detailed exception then send the user a 500 about query_rewrite failing.